### ISSUE TITLE
Work Package Templates - Edit Endpoint

### ISSUE THREAD

**Issue by RChandler234** [Northeastern-Electric-Racing/FinishLine#2150]
### Description

We need an endpoint so that users can edit work package templates in the database. 

### Acceptance Criteria

- [x] Add all these things in the work package files
- [x] This endpoint should be on the route '/template/:workpackageTemplateId/edit'
- [x] The payload should be:
        templateName String
        templateNotes String
        workPackageName String?
        stage  Work_Package_Stage?
        duration  Int?
        expectedActivities String[]
        deliverables String[]
        blockedBy BlockedByInfo[]

      BlockedByInfo should contain a name and optionally a stage (work package stage)
        
- [x] Only admins should be able to edit work package templates
- [x] We should verify that the blocked by info corresponds to an existing work package template entry through the name and stage fields (the idea is that there will be a series of templates that link to each other via blocked by)
- [x] Backend tests will be added in another ticket

### Proposed Solution

Look at other create endpoints for an example
